src/builder/create_interaction_response.rs

```diff
@@ -160,13 +163,13 @@ impl CreateInteractionResponseData {
 
     /// Adds an embed to the message.
```

Comment:
Should probably add a paragraph like "If you're calling this function while editing a message, existing embeds on the message are cleared" to make it clear what happens to existing embeds (phrasing up for debate)

Reply:
But doesn't it not clear the existing embeds?
`.add_embed` and `.add_embeds` just add additional embeds
`.embed` and `.embeds` clear existing embeds, and they already document that

Reply:
Say you posted a message with embed A. Editing without calling add_embed() will keep embed A. Editing with calling add_embed(B) will replace the embed with B. Editing with calling add_embed(B).add_embed(C) will replace the embed with B and C

Reply:
Discord interprets `None` in the embeds field as "no change" and `Some()` as "replace all existing embeds with these". That's why embeds were previously always cleared; there was no Option so Discord always received Some

Reply:
Ah ok i got it, added them

Reply:
Feel free to improve the phrasing. It wasn't clear to you so it won't be clear to many others as well

Reply:
It's it's best to reword the comment into future tense, and remove the use of 2nd-person. E.g, "calling this while editing..."

Reply:
I updated it to 
```rs
/// Calling this while editing a message will overwrite existing embeds.
```